src/model/CoilSystemCoolingDXHeatExchangerAssisted.cpp

```diff
@@ -252,6 +254,24 @@ namespace model {
     setHeatExchanger(heatExchanger);
   }
 
+  CoilSystemCoolingDXHeatExchangerAssisted::CoilSystemCoolingDXHeatExchangerAssisted(const Model& model, const AirToAirComponent& heatExchanger)
+    : StraightComponent(CoilSystemCoolingDXHeatExchangerAssisted::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::CoilSystemCoolingDXHeatExchangerAssisted_Impl>());
+
+    CoilCoolingDXSingleSpeed coolingCoil(model);
+    setCoolingCoil(coolingCoil);
+
+    if (heatExchanger.optionalCast<HeatExchangerAirToAirSensibleAndLatent>()) {
+      // no-op
+    } else if (heatExchanger.optionalCast<HeatExchangerDesiccantBalancedFlow>()) {
+      // no-op
+    } else {
+      LOG_AND_THROW("AirToAirComponent type '" << heatExchanger.briefDescription()
+                                               << "' not currently supported as a heat exchanger type for CoilSystemCoolingDXHeatExchangerAssisted.");
+    }
```

Comment:
I think for this block you could rely on IDD object-list instead of harcoding. It would be more future proof too.
```
bool ok = setHeatExchanger(heatExchanger);
if (!ok) {
  LOG_AND_THROW("Unable to set " << briefDescription() << "'s Heat Exchanger " << heatExchanger.briefDescription() << ".");
}
```